src/Enum.php

```diff
@@ -277,12 +296,15 @@ protected function resolveByStaticCall(): array
             throw new InvalidValueException($name, static::class);
         }
 
-        return static::resolve()[strtoupper($name)];
+        $resolved = static::resolve()[strtoupper($name)];
+        $resolved['name'] = strtoupper($name);
```

Comment:
Could we add this `name` property assignment in `resolve` method instead of here?

```diff
self::$cache[$class][$name]['value'] = static::getMappedValue($name) ?? static::make($name)->getValue();
self::$cache[$class][$name]['index'] = static::getMappedIndex($name) ?? static::make($name)->getIndex();
+self::$cache[$class][$name]['name'] = $name;
```

Reply:
Yeah, it's already there as key - but yes we could do it.

Reply:
If it means that this the line with `$resolved['name'] = strtoupper($name);` can be ditched, I think it would be a better solution — I also think `EnumDefinition` would have benefits in this case btw

Reply:
https://github.com/spatie/enum/pull/25/commits/9f613c960e63657e94f9cac868cc8053bd634401